primitives/runtime/src/lib.rs

```diff
@@ -246,7 +246,9 @@ impl traits::IdentifyAccount for MultiSigner {
 		match self {
 			MultiSigner::Ed25519(who) => <[u8; 32]>::from(who).into(),
 			MultiSigner::Sr25519(who) => <[u8; 32]>::from(who).into(),
-			MultiSigner::Ecdsa(who) => sp_io::hashing::blake2_256(who.as_ref()).into(),
+			MultiSigner::Ecdsa(who) => sp_io::hashing::blake2_256(
+				&who.as_compressed().expect("what should we do?")[..],
```

Comment:
@gavofyork what should we do here? I think the best would be to propagate the error via `IdentifyAccount`.

Reply:
i believe this cannot fail for any valid keypair, so `.expect` should be fine